codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/GlacierAccountIdMiddleware.kt

```diff
@@ -0,0 +1,41 @@
+package software.amazon.smithy.aws.swift.codegen.middleware
+
+import software.amazon.smithy.codegen.core.SymbolProvider
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.swift.codegen.ClientRuntimeTypes
+import software.amazon.smithy.swift.codegen.ServiceGenerator
+import software.amazon.smithy.swift.codegen.SwiftTypes
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.middleware.MiddlewarePosition
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareRenderable
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareRenderableExecutionContext
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareStep
+import software.amazon.smithy.swift.codegen.model.expectShape
+
+class GlacierAccountIdMiddleware() : MiddlewareRenderable {
+    override val name = "GlacierAccountIdAutoFill"
+
+    override val middlewareStep = MiddlewareStep.INITIALIZESTEP
+
+    override val position = MiddlewarePosition.AFTER
+
+    override fun render(model: Model, symbolProvider: SymbolProvider, writer: SwiftWriter, op: OperationShape, operationStackName: String, executionContext: MiddlewareRenderableExecutionContext) {
+        val outputShapeName = ServiceGenerator.getOperationOutputShapeName(symbolProvider, model, op)
+        val outputErrorShapeName = ServiceGenerator.getOperationErrorShapeName(op)
+        val accountId = model.expectShape<StructureShape>(op.input.get()).members().first { it.memberName.lowercase() == "accountid" }
+        writer.openBlock(
+            "$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, id: \"${name}\") { (context, input, next) -> \$N<\$N<$outputShapeName>, \$N<$outputErrorShapeName>> in", "}",
+            SwiftTypes.Result,
+            ClientRuntimeTypes.Middleware.OperationOutput,
+            ClientRuntimeTypes.Core.SdkError
+        ) {
+            writer.write("var copiedInput = input")
+            writer.openBlock("if input.${accountId.memberName}.isNilOrEmpty {", "}") {
+                writer.write("copiedInput.${accountId.memberName} = \"-\"")
+            }
+            writer.write("return next.handle(context: context, input: copiedInput)")
```

Comment:
I think we should avoid creating a copy of the input if we don't need it.  Therefore, i think the generated code would look like this:
```
            if input.accountId.isNilOrEmpty {
                var copiedInput = input
                copiedInput.accountId = "-"
                return next.handle(context: context, input: copiedInput)
            }
            return next.handle(context: context, input: input)
```

That being said, I don't like the idea of having to expose (and publically support a function called  `isNilOrEmpty`), therefore, seems like it would be better just to render:
```
            guard let accountId = input.accountId, !accountId.isEmpty else {
                var copiedInput = input
                copiedInput.accountId = "-"
                return next.handle(context: context, input: copiedInput)
            }
            return next.handle(context: context, input: input)
```

Reply:
k will make that change and just remove that function then since we don't need it.